src/typedRequestData.utils.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  MessageTypeProperty,
+  MessageTypes,
+  SignTypedDataVersion,
+  TypedDataUtils,
+  TypedMessage,
+} from '@metamask/eth-sig-util';
+import type { EnvelopingTypes } from '@rsksmart/rif-relay-contracts/dist/typechain-types/contracts/RelayHub';
+
+export type EIP712Domain = TypedMessage<EnvelopingMessageTypes>['domain'];
+
+export const eip712DomainType = [
+  { name: 'name', type: 'string' },
+  { name: 'version', type: 'string' },
+  { name: 'chainId', type: 'uint256' },
+  { name: 'verifyingContract', type: 'address' },
+];
+
+export const relayDataType = [
+  { name: 'gasPrice', type: 'uint256' },
+  { name: 'feesReceiver', type: 'address' },
+  { name: 'callForwarder', type: 'address' },
+  { name: 'callVerifier', type: 'address' },
+];
+
+export const requestType = [
+  { name: 'relayHub', type: 'address' },
+  { name: 'from', type: 'address' },
+  { name: 'to', type: 'address' },
+  { name: 'tokenContract', type: 'address' },
+  { name: 'value', type: 'uint256' },
+  { name: 'nonce', type: 'uint256' },
+  { name: 'tokenAmount', type: 'uint256' },
+  { name: 'tokenGas', type: 'uint256' },
+  { name: 'data', type: 'bytes' },
+  { name: 'relayData', type: 'RelayData' },
+];
+
+export const relayRequestType = [
+  ...relayDataType,
+  { name: 'gas', type: 'uint256' },
+];
+
+export const deployRequestType = [
+  ...relayDataType,
+  { name: 'recoverer', type: 'address' },
+  { name: 'index', type: 'uint256' },
+];
+
+export type EnvelopingMessageTypes = MessageTypes & {
+  RelayRequest: MessageTypeProperty[];
+  RelayData: MessageTypeProperty[];
+};
+
+export function getDomainSeparator(
+  verifyingContract: string,
+  chainId: number
+): EIP712Domain {
+  return {
+    name: 'string name,string version',
+    version: '2',
+    chainId: chainId,
+    verifyingContract: verifyingContract,
+  };
+}
```

Comment:
```suggestion
function getDomainSeparator(
  verifyingContract: string,
  chainId: number
): EIP712Domain {
  return {
    name: domainSeparatorType.name,
    version: domainSeparatorType.version,
    chainId: chainId,
    verifyingContract: verifyingContract,
  };
}
```

Reply:
The domain separator structure includes a name and version, but you are putting the structure and it must match with the one inside the contracts:

https://github.com/rsksmart/rif-relay-contracts/blob/master/contracts/factory/SmartWalletFactory.sol#L84
https://github.com/rsksmart/rif-relay-contracts/blob/master/contracts/smartwallet/SmartWallet.sol#L27

Reply:
First, my bad for not checking the code I copied over from https://github.com/rsksmart/rif-relay-contracts/blob/master/types/EIP712/TypedRequestData.ts#L82 (I would not likely have written
`{ ..., chainId: chainId, verifyingContract: verifyingContract }`).
Second, I really don't know what you're trying to convey here, sorry. Please rephrase.

Reply:
In your code the domain separator is wrong. It should be like this:

```suggestion
export function getDomainSeparator(
  verifyingContract: string,
  chainId: number
): EIP712Domain {
  return {
    name: 'RSK Enveloping Transaction',
    version: '2',
    chainId: chainId,
    verifyingContract: verifyingContract,
  };
}
```

Reply:
that's just repeating yourself.
but actually I do see now what you mean. I don't know how the name parameter managed to get changed like that...

Reply:
done